packages/tocco-ui/src/utilStyles/declareInteractionColors.js

```diff
@@ -23,57 +31,150 @@ const declareInteractionColors = (colors, format = stylingFormat.HTML) => {
+    activeBackground: bg[2]
   }
 }
 
-const declareFlatPrimaryColors = props => {
+const generateFlatPrimaryColors = props => {
+  const primary = themeGet('colors.primary', '#010101')(props)
+  const paper = themeGet('colors.paper', '#fefefe')(props)
+  const fg = generateInteractionColor(primary, {
+    action: getLuminance(paper) > 0.5 ? 'darken' : 'lighten'
+  })
+  const bg = generateInteractionColor(paper)
+
   return {
-    defaultColor: theme('colors.primary.line.0')(props),
-    defaultBackground: theme('colors.base.paper')(props),
-    focusColor: theme('colors.primary.line.1')(props),
-    focusBackground: theme('colors.base.fill.0')(props),
-    activeColor: theme('colors.primary.line.2')(props),
-    activeBackground: theme('colors.base.fill.1')(props)
+    defaultColor: fg[0],
+    defaultBackground: bg[0],
+    focusColor: fg[1],
+    focusBackground: bg[1],
+    activeColor: fg[2],
+    activeBackground: bg[2]
   }
 }
 
-const declareRaisedBaseColors = props => {
+const generateRaisedBaseColors = props => {
+  const text = themeGet('colors.text', '#010101')(props)
+  const paper = themeGet('colors.paper', '#fefefe')(props)
+  const fg = generateInteractionColor(text, {
+    action: getLuminance(paper) > 0.5 ? 'darken' : 'lighten'
+  })
+  const bg = generateInteractionColor(paper, {
+    shadeOffset: 0.1
+  })
+
   return {
-    defaultColor: theme('colors.base.line.0')(props),
-    defaultBackground: theme('colors.base.fill.0')(props),
-    focusColor: theme('colors.base.line.1')(props),
-    focusBackground: theme('colors.base.fill.1')(props),
-    activeColor: theme('colors.base.line.2')(props),
-    activeBackground: theme('colors.base.fill.2')(props)
+    defaultColor: fg[0],
+    defaultBackground: bg[0],
+    focusColor: fg[1],
+    focusBackground: bg[1],
+    activeColor: fg[2],
+    activeBackground: bg[2]
   }
 }
 
-const declareRaisedPrimaryColors = props => {
+const generateRaisedPrimaryColors = props => {
+  const text = themeGet('colors.text', '#010101')(props)
+  const paper = themeGet('colors.paper', '#fefefe')(props)
+  const primary = themeGet('colors.primary', '#010101')(props)
+  const higherContrast = getContrast(text, primary) > getContrast(paper, primary) ? text : paper
+  const fg = generateInteractionColor(higherContrast, {
+    action: getLuminance(primary) > 0.5 ? 'darken' : 'lighten'
+  })
+  const bg = generateInteractionColor(primary)
+
   return {
-    defaultColor: theme('colors.primary.fillContrast.0')(props),
-    defaultBackground: theme('colors.primary.fill.0')(props),
-    focusColor: theme('colors.primary.fillContrast.1')(props),
-    focusBackground: theme('colors.primary.fill.1')(props),
-    activeColor: theme('colors.primary.fillContrast.2')(props),
-    activeBackground: theme('colors.primary.fill.2')(props)
+    defaultColor: fg[0],
+    defaultBackground: bg[0],
+    focusColor: fg[1],
+    focusBackground: bg[1],
+    activeColor: fg[2],
+    activeBackground: bg[2]
```

Comment:
loads of repetitive code in this file.

Example why not make something like

```
const idk = fg => ({
   defaultColor: fg[0],
    defaultBackground: bg[0],
    focusColor: fg[1],
    focusBackground: bg[1],
    activeColor: fg[2],
    activeBackground: bg[2]
})
```

and then this 3 times:

`return idk(fg)  `


what does fg stand for?

Reply:
fg = foreground
bg = background